src/tests/GC/API/Frozen/Frozen.cs

```diff
@@ -134,16 +134,17 @@ internal class Node
 
     internal static class Program
     {
-        private static unsafe IntPtr GetMethodTablePointer(object obj)
+        internal sealed class RawData
         {
-            GCHandle gch = GCHandle.Alloc(obj);
-            IntPtr pointerToPointerToObject = GCHandle.ToIntPtr(gch);
-            IntPtr pointerToObject = *((IntPtr*)pointerToPointerToObject);
-            IntPtr methodTable = *((IntPtr*)pointerToObject);
-            gch.Free();
-            return methodTable;
+            public byte Data;
         }
 
+        internal static ref byte GetRawData(this object obj) =>
+            ref Unsafe.As<RawData>(obj).Data;
+
+        internal static unsafe IntPtr GetMethodTablePointer(object obj) =>
+            (IntPtr)Unsafe.Add(ref Unsafe.As<byte, IntPtr>(ref obj.GetRawData()), -1);
```

Comment:
This assumes that offset 0 is a valid byref. It happens to be true in CoreCLR today, but it is not a safe assumption to make.